src/main.cpp

```diff
@@ -6502,9 +6509,13 @@ bool SendMessages(CNode* pto, CConnman& connman)
     const Consensus::Params& consensusParams = Params().GetConsensus();
     {
         // Don't send anything until we get its version message
-        if (pto->nVersion == 0)
+        if (!pto->fSuccessfullyConnected || pto->fDisconnect)
```

Comment:
I think if we are still referencing version here to explain why,  then `fSuccessfullyConnected` isn't appropriate.